app/web_framework/cluster_master_application.py

```diff
@@ -296,7 +275,8 @@ class _SlavesHandler(_ClusterMasterBaseAPIHandler):
     def post(self):
         slave_url = self.decoded_body.get('slave')
         num_executors = int(self.decoded_body.get('num_executors'))
-        response = self._cluster_master.connect_slave(slave_url, num_executors)
+        session_id = self.decoded_body.get('session_id')
```

Comment:
Yep, `get` will return None so this should be backward compatible (the master would work with old slave versions). That should also be true with the next change that adds session id validation.

I just realized we should probably do more validation on this api data though. I'll add a todo.